cohortextractor/therapeutics_utils.py

```diff
@@ -0,0 +1,37 @@
+import re
+from pathlib import Path
+
+
+def extract_risk_groups_from_file(input_file):
+    """
+    Helper function to extract distinct risk groups from a file with one line
+    per distinct risk group value extracted from the database
+    """
+    lines = Path(input_file).read_text().split("\n")
+    risk_groups = set()
+    for line in lines:
+        groups = line.split("and")
+        for group in groups:
+            if group:
```

Comment:
Is this because `group == False` when `line == ""`? If so, and the empty lines are at the start or end of the input file, then might this be clearer:

```python
lines = Path(input_file).read_text().strip().split("\n")
```

I'm not yet clear where `input_file` comes from, so sorry for what is probably unnecessary pedantry 🧐.

Reply:
Hmm, I'll check.
`input_file` is just a manually generated text file with distinct values from the dataset risk group columns.  Basically, Helen ran a query in the server and copy-pasted the output into a file (after checking that there's nothing there that looks like it came from a free text field and could be disclosive), but it consists of lots of combinations e.g. "liver disease and IMID and solid cancer" etc.  The plan is that someone will run that query when the data changes, and we'll update the `ALLOWED_RISK_GROUPS` with it.  I doubt it'll change often, because the list currently has all the conditions specific on the form checkboxes, but if we do get free text data there could a risk group condition listed there that we want to capture.